pdns/dnsdist-cache.cc

```diff
@@ -294,75 +294,80 @@ bool DNSDistPacketCache::get(DNSQuestion& dq, uint16_t queryId, uint32_t* keyOut
 
 /* Remove expired entries, until the cache has at most
    upTo entries in it.
+   If the cache has more than one shard, we will try hard
```

Comment:
Is this true? I think if no expired entries are in a shard, nothing will be deleted...

Reply:
We will only remove expired entries, yes, we are in `purgeExpired()`, but we will scan all entries if we need to.